lib/payment_manager.ts

```diff
@@ -0,0 +1,55 @@
+import * as BigNumber from 'bignumber.js'
+import ChainManager from './chain_manager'
+import { PaymentChannel } from './payment_channel'
+import Payment from './payment'
+import ChannelContract from './channel_contract'
+
+export default class PaymentManager {
+  private chainManager: ChainManager
+
+  private channelContract: ChannelContract
+
+  constructor (chainManager: ChainManager, channelContract: ChannelContract) {
+    this.chainManager = chainManager
+    this.channelContract = channelContract
+  }
+
+  async buildPaymentForChannel (channel: PaymentChannel, value: BigNumber.BigNumber, price: BigNumber.BigNumber, meta: string): Promise<Payment> {
+    const totalValue = channel.spent.plus(value)
+    const digest = await this.channelContract.paymentDigest(channel.channelId, totalValue)
+    const signature = await this.chainManager.sign(channel.sender, digest)
+
+    return new Payment({
+      channelId: channel.channelId,
+      sender: channel.sender,
+      receiver: channel.receiver,
+      price,
+      value: totalValue,
+      channelValue: channel.value,
+      signature,
+      meta,
+      contractAddress: channel.contractAddress,
+      token: undefined
+    })
+  }
+
+  async isValid (payment: Payment, paymentChannel: PaymentChannel): Promise<boolean> {
```

Comment:
Note that this method was unused in the original implementation - I included it here because there was a `TODO` attached to it saying that it should be used. I can create a separate PR to use this method.